src/server/pkg/dbutil/dbutil.go

```diff
@@ -0,0 +1,65 @@
+package dbutil
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jinzhu/gorm"
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+)
+
+// set this to true if you want to keep the database around
+var devDontDropDatabase = false
+
+const (
+	DefaultPostgresHost = "127.0.0.1"
+	DefaultPostgresPort = 32228
+	TestPostgresUser    = "postgres"
+)
+
+// WithTestDB connects to postgres using the default settings, creates a database with a unique name
+// then calls cb with a sqlx.DB configured to use the newly created database.
+// After cb returns the database is dropped.
+func WithTestDB(t testing.TB, cb func(db *sqlx.DB)) {
+	dsn := fmt.Sprintf("host=%s port=%d user=%s sslmode=disable", DefaultPostgresHost, DefaultPostgresPort, TestPostgresUser)
+	db := sqlx.MustOpen("postgres", dsn)
+	dbName := fmt.Sprintf("test_%d", time.Now().UnixNano())
+	db.MustExec("CREATE DATABASE " + dbName)
+	t.Log("database", dbName, "successfully created")
+	db2 := sqlx.MustOpen("postgres", dsn+" dbname="+dbName)
+	cb(db2)
```

Comment:
We probably want the testing object passed into the callback.

Reply:
I don't think that sort of threading through is necessary. I can't foresee a situation where it would be useful to modify the testing object or produce another one and then inject it into the callback.  The reason we pass it into the outer function is to make test failures during setup easier to debug. (they create failed assertions and short circuit the test, just like everything else)